open_discussions/authentication_test.py

```diff
@@ -0,0 +1,57 @@
+"""Tests for authentication"""
+from open_discussions.authentication import (
+    get_encoded_and_signed_subscription_token,
+    decode_and_verify_username_from_token,
+    StatelessTokenAuthentication,
+)
+
+
+def test_get_encoded_and_signed_subscription_token(user):
+    """Tests that get_encoded_and_signed_subscription_token returns a token"""
+    assert get_encoded_and_signed_subscription_token(user) is not None
+
+
+def test_decode_and_verify_username_from_token(user):
+    """Tests that decode_and_verify_username_from_token returns the encoded username"""
+    token = get_encoded_and_signed_subscription_token(user)
+    assert decode_and_verify_username_from_token(token) == user.username
+
+
+def test_stateless_token_authentication_expired(rf):
+    """Tests that StatelessTokenAuthentication returns None if token is expired"""
+    token = 'MDFDNEZBV1dFOTc1VDFLRk4wQ0RZNllLUFo6MWVvRTlrOi05V2VacjYxWm9aR25wdTdVUlY2RGFQUXRRSQ=='
+    request = rf.get('api/v0/notification_settings', HTTP_AUTHORIZATION='Token {}'.format(token))
+    authentication = StatelessTokenAuthentication()
+    assert authentication.authenticate(request) is None
```

Comment:
maybe add a similar, sanity-check type test that a nonsense token value (e.g. `mary had a little lamb`) also returns None?

Reply:
I had to update the exception handling in `decode_and_verify_username_from_token` to a bare except since a garbase token still base64 decodes but blows up with a unicode error during unsigning.